be/src/exec/pipeline/pipeline_driver_executor.cpp

```diff
@@ -459,7 +462,7 @@ RuntimeProfile* GlobalDriverExecutor::_build_merged_instance_profile(QueryContex
         merged_driver_profiles.push_back(merged_driver_profile);
     }
 
-    new_instance_profile = query_ctx->object_pool()->add(new RuntimeProfile(instance_profile->name()));
+    new_instance_profile = obj_pool->add(new RuntimeProfile(instance_profile->name()));
```

Comment:
It will be better to return a std::unique_ptr<RuntimeProfile> and let the client decide how to use the returned object.

Reply:
this method will be invoked recursively. Seems cannot be formed that way.